### DISCUSSION TITLE
Want to register two different types input fields under same name

### DISCUSSION THREAD

**Discussion by usmanshah9** [react-hook-form/react-hook-form#7598] [Q&A]
I am creating a dynamic for from json data and everything is working fine except for this one issues, I am trying to register two radio buttons and an text input under the same name hoping to get the final value as an object. 
Tried using this syntax
```
   <Form.Check
            label='Yes'
            {...register(`${fieldData.text}.Boolean`, {
              onChange: (e) => {
                handleCheckBox(
                  e.target.value,
                  fieldData,
                  fieldData.proceedError
                );
              },
            })}
            type='radio'
            value='yes'
          />
 ```

I am getting the following error:
TypeError: Cannot read properties of undefined (reading '_f')
Can someone please offer a solution to this problem, I am stuck in my progression right at this spot and I'd be grateful if you could help me solve this in the correct way.

**Comment by bluebill1049** [comment]
not possible with two inputs with the same name (react hook form).

**Reply by usmanshah9** [reply]
Thank for the quick response bill, 
Can you suggest some solution for my problem,
The nature of my form element if
### Question
o yes
o no
### If yes
Answer in a text input

I want to store values of this element as an object, is there any way I can achieve that using RHF.

**Reply by bluebill1049** [reply]
`register('objectData', { value: { object: '1', object1: '2' } })`

or just use `useController`

**Reply by usmanshah9** [reply]
I am actually rendering out multiple questions like above via array.map, so the names are dynamically generated.

**Reply by bluebill1049** [reply]
````
// array 
register('objectData.0')
register('objectData.1')

// object
register('objectData.test')
register('objectData.test1')
````

please refer to the doc under `register` page

**Reply by usmanshah9** [reply]
I feel really dumb now as I cant really deliver my question correctly :/,
but I am using this method to register and the filed name comes from the object from array.map as fieldData.text but when I try to register it using this syntax {...register(`${fieldData.text}.0`} it gives error like following

TypeError: Cannot read properties of undefined (reading '_f')

Maybe its a syntax error or something else , this is my original problem.

**Reply by bluebill1049** [reply]
reproduce that in a codesandbox.

**Reply by usmanshah9** [reply]
This is exactly how I render this field in my form, this is with a static object data
https://codesandbox.io/s/friendly-keller-fs4yr?file=/src/App.js
But this works, doesn't work when the data dynamically comes from parent component.

**Comment by Moshyfawn** [comment]
Is [this CSB](https://codesandbox.io/s/dynamic-radio-example-2cr9k) something you're kinda looking for?
P.S. I left out all the boolean conversions, JSON implementations and input field conditions for the sake of simplicity, because I don't k now your JSON form structure.

**Reply by usmanshah9** [reply]
Hi, there thanks for the help,
This is my [CBS](https://codesandbox.io/s/friendly-keller-fs4yr?file=/src/App.js)
But now this is working exactly how I want :/,
I dont know why I'm getting the error when the fieldData comes from array.map

**Reply by Moshyfawn** [reply]
[You CSB](https://codesandbox.io/s/friendly-keller-fs4yr?file=/src/App.js) is empty :P
<img width="1388" alt="image" src="https://user-images.githubusercontent.com/16290753/150325973-f0cca8d6-f00f-47bd-b229-90a00497f357.png">

**Reply by usmanshah9** [reply]
My bad, check now. I dont really use CSB that much

**Reply by Moshyfawn** [reply]
[This is your CSB](https://codesandbox.io/s/falling-cdn-1zfxq?file=/src/App.js:498-502) with emulated remote data fetching. Works fine for me. Can you reproduce the behavior that doesn't work for you?

**Reply by usmanshah9** [reply]
Yeah, I'm already working on it :)

**Reply by usmanshah9** [reply]
Damn even the [recreated one ](https://codesandbox.io/s/friendly-keller-fs4yr?file=/src/App.js)is working perfectly fine :/

I'm doing something very stupid in my code I suppose 💯

**Reply by Moshyfawn** [reply]
I'm not sure we can help if it's not reproducible :P
Check for question field names in your not working use-case, what's passed as the field name, key. Bracket vs dot notations.
The ` TypeError: Cannot read properties of undefined (reading '_f')` issue is usually due to some mismatch of dynamic field names provided

**Reply by usmanshah9** [reply]
Thank you for your help <3

**Reply by Moshyfawn** [reply]
Also, make sure you don't have special characters passed as a field name. See  #7494

**Reply by usmanshah9** [reply]
I'll let you know what was the problem :)

**Reply by usmanshah9** [reply]
Bro its strange that If i slice my name string from 0 to 54 it somehow works. Is there a character limit on the name of register field?

**Reply by usmanshah9** [reply]
Or there might be a special character on the 55th index on some of the quesitions....

**Reply by Moshyfawn** [reply]
> Is there a character limit on the name of register field

There shouldn't be any character limit

> Or there might be a special character on the 55th index on some of the quesitions....

If that's the case, then you should consider using simpler object keys

**Reply by bluebill1049** [reply]
> Bro its strange that If i slice my name string from 0 to 54 it somehow works. Is there a character limit on the name of register field?

there is no limitation on the index number
